soc/st/stm32/Kconfig

```diff
@@ -56,17 +56,42 @@ config STM32_WKUP_PINS
 	help
 	  Enable support for STM32 PWR wake-up pins.
 
+# On STM32WB0, the default selection is 'SMPS enabled'
+# as it corresponds to reset state and works regardless of
+# hardware configuration, whereas disabling SMPS on boards
+# using the SMPS hardware configuration will "brick".
 choice POWER_SUPPLY_CHOICE
 	prompt "STM32 power supply configuration"
-	default POWER_SUPPLY_LDO
+	default POWER_SUPPLY_LDO if !SOC_SERIES_STM32WB0X
+	default POWER_SUPPLY_SMPS_SUPPLIES_LDO if SOC_SERIES_STM32WB0X
```

Comment:
Moving to `soc/st/stm32/stm32wb0x/Kconfig` appears to work. Will do the necessary changes.

However, since other power supply configuration is being moved to DTS, should I take the opportunity to also move this to DTS? (for WB0 only)

Reply:
If not too much trouble, why not. But not mandatory if this extends the work too much.